### ISSUE TITLE
Ability to put site into maintenance mode

### ISSUE THREAD

**Issue by McFlat** [Chocobozzz/PeerTube#1230]
We need a way to run a command so to make the site go into maintenance mode and display a message to users that the site is currently in maintenance mode and to please check back after a specified interval set by the admin as to the duration for the maintenance window.

It would be a requirement say to do backup of the database or to switch the site to a different server, so that as admin is doing those tasks data in the database doesn't keep changing or new videos don't start transcoding etc. This way it would allow an admin to make a complete accurate backup of the site and transition it to another server or update the database or other software on the server without worrying that data will be lost in the process.

Perhaps a tool script would be nice for this, so that admin can run it from a server, then it creates a entry in the database or creates a file on the server(possibly this would only work for single server deployment, and doesn't take into account multiple server nodes behind a load balancer) and then it would show the configured message to users that the site is in maintenance and to check back in an hour or a half hour or couple hours, whatever the amount of time is specified by the admin.

**Comment by Nutomic** [comment]
Instead of specifying a time interval, it might be better if the admin can just set a message that is displayed to users (which can include a time of course).

**Comment by McFlat** [comment]
@Nutomic Yeah that would be better, instead of using a template that fills in the blanks and one of the blanks is a time it could just be a text string with any message so users know to come back soon.

**Comment by Chocobozzz** [comment]
Stop peertube, and put a custom static file in nginx would be a good start I think

**Comment by McFlat** [comment]
Yeah that would do it, but I think there needs to be a command that does it for the admin, automatically. Then when all is ready the admin can run a command to take the site out of maintenance mode and it would remove the static html file from nginx.

**Comment by scanlime** [comment]
I'd really find it useful if the "maintenance mode" responder, whatever that may be, can be easily run on a different piece of hardware. My ideal scenario might be a script that exports any necessary config info (and https private keys?) to a raspberry pi or other backup server.

When I've performed physical server maintenance in the past, I've just left the server offline to time out. I didn't want to return 404s nor did I want to audit the code to see which response codes might have been safe.

**Comment by scanlime** [comment]
In my case the maintenance server could be at the same public IP as the production server, but of course people will have a variety of network setups.
